tweetbot.go

```diff
@@ -36,56 +38,93 @@ func check(e error) {
 func WriteToFile(filename string, data string) error {
 	file, err := os.Create(filename)
 	if err != nil {
-		return err
+		return fmt.Errorf("could not open file: %s", err)
 	}
-	defer file.Close()
+	defer closeFile(file)
 
 	_, err = io.WriteString(file, data)
-	check(err)
-	return file.Sync()
+	if err != nil {
+		return fmt.Errorf("could not write string: %s", err)
+	}
+	return nil
+}
+
+func getLastTweetID() (int64, error) {
+	lastTweetIdFile, err := os.Open("lastTweetId")
+	if err != nil {
+		return 0, fmt.Errorf("could not open lastTweetId: %s", err)
+	}
+	defer closeFile(lastTweetIdFile)
+
+	lastTweetId, err := ioutil.ReadAll(lastTweetIdFile)
+	if err != nil {
+		return 0, fmt.Errorf("could not read lastTweetId content: %s", err)
+	}
+
+	lastTweetIdValue, err := strconv.ParseInt(string(lastTweetId), 10, 64)
+	if err != nil {
+		return 0, fmt.Errorf("could not parse lastTweetId content as int64: %s", err)
+	}
+
+	return lastTweetIdValue, nil
 }
 
 func main() {
 
 	file, err := os.Open("settings.yaml")
 	if err != nil {
-		panic(err)
+		log.Fatalf("could not open file: %s", err)
 	}
+	defer closeFile(file)
 
 	filecontent, err := ioutil.ReadAll(file)
-	check(err)
+	if err != nil {
+		log.Fatalf("could not read settings.yaml content: %s", err)
+	}
 
 	var conf Configuration
 	err = yaml.Unmarshal(filecontent, &conf)
-	check(err)
-
-	fileReference, err := os.Open("lastTweetId")
-	check(err)
+	if err != nil {
+		log.Fatalf("could not unmarshal settings.yaml content: %s", err)
+	}
 
-	lastTweetId, err := ioutil.ReadAll(fileReference)
-	check(err)
+	var lastTweetID int64
+	if id, err := getLastTweetID(); err != nil {
+		log.Printf("could not get last tweet id: %s", err)
+	} else {
+		lastTweetID = id
+	}
```

Comment:
If the lastTweetId file is not present everything would stop but ideally we continue and create the file after the first run